### PR TITLE
HW7: broker & logger

### PR DESCRIPTION
**Pull request by andrii0yerko**
- Implement logger abstraction for HTTP request monitoring
- Implement a logger that produces RabbitMQ events
- Implement a CLI tool for log monitoring

### CODE REVIEW

File: pkg/logger/rabbitmq.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+type RabbitMQLogger struct {
+	conn       *amqp.Connection
+	channel    *amqp.Channel
+	exchange   string
+	routingKey string
+	queueName  string
+}
+
+func NewRabbitMQLogger(amqpURI, exchange, routingKey, queueName string) (*RabbitMQLogger, error) {
+	conn, err := amqp.Dial(amqpURI)
```
Comment (nazarsavorona) [inline pkg/logger/rabbitmq.go:20]: I'd create the connection outside, so it can be shared.

File: cmd/log-monitoring/main.go
Comment (nazarsavorona) [inline cmd/log-monitoring/main.go:1]: Nice approach. No need to create flags or other crap like I did. 😄

File: cmd/log-monitoring/main.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+		false,
+		false,
+		nil,
+	)
+	if err != nil {
+		log.Printf("Failed to consume messages: %s", err)
+	}
+
+	log.Println("Waiting for messages. Press Ctrl+C to exit.")
+	signals := make(chan os.Signal, 1)
```
Comment (nazarsavorona) [inline cmd/log-monitoring/main.go:68]: Nice approach.

File: pkg/logger/console.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+func (l *ConsoleLogger) Info(message string) {
+	l.logMessage(logLevelInfo, message)
+}
+
+func (l *ConsoleLogger) Error(message string) {
+	l.logMessage(logLevelError, message)
+}
+
+func (l *ConsoleLogger) logMessage(level string, message string) {
```
Comment (nazarsavorona) [inline pkg/logger/console.go:27]: Looks like you only need this method. Perhaps, somebody would agree that having three methods is more convenient. Depends on a goal.

**Review by nazarsavorona — APPROVED** [review]
Left a few comments. Overall, good job, well done.

File: cmd/log-monitoring/main.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/streadway/amqp"
+)
+
+func main() {
+	amqpURI := "amqp://admin:admin@localhost:5672/"
```
[the remark this answers is not in the document]
Comment (andrii0yerko) [reply cmd/log-monitoring/main.go:14]: oh yes 😅
Have no time to make this util configurable too

But you're right, it should be

**Pull request merged**